deepforest/main.py

```diff
@@ -382,13 +377,23 @@ def predict_file(self, csv_file, root_dir, savedir=None, color=None, thickness=1
         Returns:
             df: pandas dataframe with bounding boxes, label and scores for each image in the csv file
         """
+        df = pd.read_csv(csv_file)
+        ds = dataset.TreeDataset(csv_file=csv_file,
+                                 root_dir=root_dir,
+                                 transforms=None,
+                                 train=False)
+        dataloader = self.predict_dataloader(ds)
+
         results = predict.predict_file(model=self,
-                                       csv_file=csv_file,
+                                       trainer=self.trainer,
+                                       annotations=df,
+                                       dataloader=dataloader,
```

Comment:
I'm assuming there's a reason related to the overarching PR for moving the dataloader and trainer creation, presumably related to handling the trainer? On the off chance there isn't I think it's simplest to move it back into `predict_file` (see main comment above)

Reply:
We want to isolate predict.predict_* methods from main.predict_* methods. If you have a csv file as the argument, you need to pass the class to the function to generate the dataloader, see above comment.

Reply:
:+1: